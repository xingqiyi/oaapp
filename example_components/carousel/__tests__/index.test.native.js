import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Carousel from '../index';

describe('Carousel.RN', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Carousel><View>abc</View></Carousel>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has no dots', () => {
    const wrapper = shallow(
      <Carousel dots={false}>
        <View>abc</View>
      </Carousel>,
    );
    // only has `ScrollView`
    expect(wrapper.children()).toHaveLength(1);
    // ScrollViewMock
    expect(wrapper.children().name()).toMatch(/ScrollView/);
  });

  it('has dots', () => {
    const wrapper = shallow(<Carousel><View>abc</View></Carousel>);
    expect(wrapper.children()).toHaveLength(2);
  });

  it('check api', () => {
    const itemStyle = {
      height: 150,
      width: 320,
      flexGrow: 1,
    };
    const wrapper = shallow(
      <Carousel selectedIndex={1}>
        <View style={itemStyle}>item1</View>
        <View style={itemStyle}>item2</View>
        <View style={itemStyle}>item3</View>
      </Carousel>,
    );
    // wrapper.children().at(0) => ScrollView(ScrollViewMock)
    expect(wrapper.children().at(0).children()).toHaveLength(3);
    expect(wrapper.state('selectedIndex')).toEqual(1);
    // because of ``ScrollViewMock``, can not test UI and events.
  });
});
