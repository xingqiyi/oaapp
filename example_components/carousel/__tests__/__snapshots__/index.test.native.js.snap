// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel.RN renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayInterval={3000}
    bounces={true}
    contentContainerStyle={undefined}
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    directionalLockEnabled={true}
    dotActiveStyle={Object {}}
    dotStyle={Object {}}
    dots={true}
    horizontal={true}
    infinite={false}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagination={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    scrollEventThrottle={100}
    selectedIndex={0}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    styles={
      Object {
        "pagination": Object {
          "alignItems": "center",
          "position": "absolute",
        },
        "paginationX": Object {
          "bottom": 10,
          "left": 0,
          "right": 0,
        },
        "paginationY": Object {
          "bottom": 0,
          "right": 10,
          "top": 0,
        },
        "pointActiveStyle": Object {
          "backgroundColor": "#888",
        },
        "pointStyle": Object {
          "backgroundColor": "#cccccc",
          "borderRadius": 8,
          "height": 8,
          "width": 8,
        },
        "spaceStyle": Object {
          "marginHorizontal": 2.5,
          "marginVertical": 3,
        },
      }
    }
    vertical={false}
  >
    <View>
      <View
        style={
          Object {
            "width": 0,
          }
        }
      >
        <View>
          abc
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "position": "absolute",
        },
        Object {
          "bottom": 10,
          "left": 0,
          "right": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#cccccc",
              "borderRadius": 8,
              "height": 8,
              "width": 8,
            },
            Object {
              "marginHorizontal": 2.5,
              "marginVertical": 3,
            },
            Object {},
            Object {
              "backgroundColor": "#888",
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
</View>
`;
